src/index.c

```diff
@@ -1169,6 +1169,9 @@ int git_index_remove_bypath(git_index *index, const char *path)
 		ret != GIT_ENOTFOUND))
 		return ret;
 
+	if (ret == GIT_ENOTFOUND)
+		giterr_clear();
+
```

Comment:
It certainly wouldn't be incorrect to `giterr_clear()` in all cases, but this does avoid setting several thread-local variables.

(Yes, I realize this is a *seriously* minor issue, but some vendors have produced code that's inferior around TLS in the past.)